yaksh/views.py

```diff
@@ -3191,3 +3191,89 @@ def download_course_progress(request, course_id):
     for student in stud_details:
         writer.writerow(student)
     return response
+
+
+@login_required
+@email_verified
+def course_forum(request, course_id):
+    user = request.user
+    course = get_object_or_404(Course, id=course_id)
+    if (not course.is_creator(user) and not course.is_teacher(user)
+            and not course.is_student(user)):
+        raise Http404('You are not enrolled in {0} course'.format(course.name))
+    posts = course.post.filter(active=True).order_by('-modified_at')
+    if request.method == "POST":
+        form = PostForm(request.POST, request.FILES)
+        if form.is_valid():
+            new_post = form.save(commit=False)
+            new_post.creator = user
+            new_post.course = course
+            new_post.save()
+            return redirect('yaksh:post_comments',
+                            course_id=course.id, uuid=new_post.uid)
+    else:
+        form = PostForm()
+    return render(request, 'yaksh/course_forum.html', {
+        'user': user,
+        'course': course,
+        'posts': posts,
+        'form': form,
+        'user': user
+        })
+
+
+@login_required
+@email_verified
+def post_comments(request, course_id, uuid):
+    user = request.user
+    post = get_object_or_404(Post, uid=uuid)
+    comments = post.comment.filter(active=True)
+    course = get_object_or_404(Course, id=course_id)
+    if (not course.is_creator(user) and not course.is_teacher(user)
+            and not course.is_student(user)):
+        raise Http404('You are not enrolled in {0} course'.format(course.name))
+    form = CommentForm()
+    if request.method == "POST":
+        form = CommentForm(request.POST, request.FILES)
+        if form.is_valid():
+            new_comment = form.save(commit=False)
+            new_comment.creator = request.user
+            new_comment.post_field = post
+            new_comment.save()
+            return redirect(request.path_info)
+    return render(request, 'yaksh/post_comments.html', {
+        'post': post,
+        'comments': comments,
+        'form': form,
+        'user': user
+        })
+
+
+@login_required
+@email_verified
+def hide_post(request, course_id, uuid):
+    user = request.user
+    course = get_object_or_404(Course, id=course_id)
+    if (not course.is_creator(user) and not course.is_teacher(user)
+            and not course.is_student(user)):
```

Comment:
You don't need `course.is_student(user)` condition as this is operation is allowed to moderator and teacher.